casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -786,16 +786,25 @@ object BlockAPI {
                          s"Could not find block with hash $hash"
                            .asLeft[Boolean]
                            .pure[F]
-                       case Some(givenBlockMetadata) =>
-                         DagOperations
-                           .bfTraverseF(List(lastFinalizedBlockMetadata)) { b =>
-                             b.parents.traverse(dag.lookup).map { parentOpts =>
-                               parentOpts.flatten.distinct
-                                 .filter(_.blockNum >= givenBlockMetadata.blockNum)
+                       case Some(givenBlockMetadata) => {
+                         // TODO this is temporary solution to not calculate fault tolerance for old blocks which is costly
+                         val oldBlock =
+                           dag.latestBlockNumber.map(_ - givenBlockMetadata.blockNum).map(_ > 100)
+                         val isInvalid           = dag.lookup(givenBlockMetadata.blockHash).map(_.get.invalid)
+                         val oldBlockIsFinalized = isInvalid.map(if (_) false else true)
```

Comment:
This can be simplified as negation `val oldBlockIsFinalized = isInvalid.not` (or`~^(isInvalid.not`)`).

Reply:
I hope to remove this code shortly anyway, and wanted to keep it similar to piece of code in faoultTolerance :)